tensorflow_similarity/architectures/resnet50.py

```diff
@@ -103,12 +106,10 @@ def ResNet50Sim(
     return SimilarityModel(inputs, outputs)
 
 
-def build_resnet(x: layers.Layer, weights: str, trainable: str) -> layers.Layer:
+def build_resnet(weights: str = None, trainable: str = "full") -> tf.keras.Model:
```

Comment:
We removed the `x: layers.Layer` here, so we'll need to update line 86 to:

`x = build_resnet(variant, weights, trainable)(x)`